Parasol/main.swift

```diff
@@ -8,5 +8,39 @@
 
 import Foundation
 
-print("Hello, World!")
+// TODO: make passing Xcode project as an optional argument
+if let xcodeProject = XcodeProject.findXcodeProjectInCurrentDirectory() {
+    let targets = xcodeProject.targets
+    print("Which target in \(xcodeProject.name):")
+    for i in 1...targets.count {
+        print("\(i)) \(targets[i - 1].name)")
+    }
+    let input = readLine()
+    if let input = input, index = Int(input) where index - 1 >= 0 && index <= targets.count  {
+        let target = targets[index - 1]
+        if case let .Exists(profdataPath, executablePath) = target.coverageDataExists {
+            for line in Coverage.showCoverageReport(profdataPath, executablePath: executablePath).lines {
+                print(line)
+            }
+        } else {
+            print("Coverage data has not been generated. Would you like to run tests? (y/n)")
+            let input = readLine()
+            if input == "y" {
+                target.runTests()
+                if case let .Exists(profdataPath, executablePath) = target.coverageDataExists {
+                    for line in Coverage.showCoverageReport(profdataPath, executablePath: executablePath).lines {
+                        print(line)
+                    }
+                } else {
+                    print("ERROR: Coverage data was not generated by Xcode.")
```

Comment:
Yes :) Every process is initialised with three open file descriptors, stdin , stdout , and stderr. Errors should go to stderr so you can extract them in logs.